### ISSUE TITLE
Colorization issues ("Struct", "Property/Setter")

### ISSUE THREAD

**Issue by seyhajin** [mutatedmonkey/sublimetext3#1]
"Struct", "Property/Setter" keywords and bodies are not colorized correctly :

![image](https://user-images.githubusercontent.com/24609692/27676665-2312f404-5caf-11e7-8bc3-2351202d4458.png)

![image](https://user-images.githubusercontent.com/24609692/27676673-2a0bdd7a-5caf-11e7-8110-9b2e64baec2d.png)

#If and #EndIf preprocessor :

![image](https://user-images.githubusercontent.com/24609692/27676783-95280a70-5caf-11e7-837d-23eb2d254084.png)

And sometimes, "Method" block but maybe cause If block without "Then". Can you have any solution for that?

**Comment by wiebow** [comment]
Yes, it's because if.. blocks are not closed properly. This is a side effect from the multiple ways of using those blocks. When you include Then in the If block statements, things should be better real soon.
